src/app/components/gv-alert/gv-alert.component.ts

```diff
@@ -0,0 +1,316 @@
+  httpStatus: Array<HttpStatus>;
+  durations: Array<number>;
+  type: string;
+  statusLabel: string;
+  responseTimeLabel: string;
+
+  get isStatusAlert(): boolean {
+    return 'status' === (this.isCreationMode ? this.alertForm?.value?.type?.toLowerCase() : this.alert?.type?.toLowerCase());
+  }
+
+  get isResponseTimeAlert(): boolean {
+    return 'response_time' === (this.isCreationMode ? this.alertForm?.value?.type?.toLowerCase() : this.alert?.type?.toLowerCase());
+  }
+
+  get isCreationMode(): boolean {
+    return this.mode === AlertMode.CREATION;
+  }
+
+  get isEditionMode(): boolean {
+    return this.mode === AlertMode.EDITION;
+  }
+
+  get isReadingMode(): boolean {
+    return this.mode === AlertMode.READING;
+  }
+
+  get isAlertEnabled(): boolean {
+    return this.alert?.enabled;
+  }
+
+  get typeLabel(): string {
+    return this.isStatusAlert ? this.statusLabel : this.responseTimeLabel;
+  }
+
+  get isAlertingEnabled() {
+    return this.status?.available_plugins > 0 && this.status?.enabled;
+  }
+
+  constructor(
+    private applicationService: ApplicationService,
+    private route: ActivatedRoute,
+    private translateService: TranslateService,
+    private notificationService: NotificationService,
+    private permissionsService: PermissionsService,
+  ) {}
+
+  ngOnInit(): void {
+    this.application = this.route.snapshot.data.application;
+    if (this.application) {
+      this.permissionsService
+        .getCurrentUserPermissions({ applicationId: this.application.id })
+        .toPromise()
+        .then((permissions) => {
+          this.permissions = permissions.ALERT;
+        });
+    }
+    this.resetAddAlert();
+    this.resetAddAlertStatus();
+    if (this.application) {
+      this.httpStatus = [
+        { value: '1xx', label: '1xx - INFORMATION' },
+        { value: '2xx', label: '2xx - SUCCESS' },
+        { value: '3xx', label: '3xx - REDIRECTION' },
+        { value: '4xx', label: '4xx - CLIENT ERROR' },
+        { value: '5xx', label: '5xx - SERVER ERROR' },
+      ].concat(HttpHelpers.httpStatus);
+      this.durations = [1, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55];
+      this.timeUnits = ['S', 'MINUTES', 'HOURS'];
+      this.translateService
+        .get([
+          i18n('application.alerts.type.status'),
+          i18n('application.alerts.type.response_time'),
+          i18n('application.alerts.timeUnits.seconds'),
+          i18n('application.alerts.timeUnits.minutes'),
+          i18n('application.alerts.timeUnits.hours'),
+        ])
+        .toPromise()
+        .then((translations) => {
+          this.statusLabel = '' + Object.values(translations)[0];
+          this.responseTimeLabel = '' + Object.values(translations)[1];
+          this.types = [
+            {
+              label: Object.values(translations)[0],
+              value: 'STATUS',
+            },
+            {
+              label: Object.values(translations)[1],
+              value: 'RESPONSE_TIME',
```

Comment:
Why not using the enum generated from the openAPI descriptor ?
See `projects/portal-webclient-sdk/src/lib/model/alertType.ts`